src/components/expanded-state/unique-token/ZoomableWrapper.js

```diff
@@ -122,12 +143,20 @@ export const ZoomableWrapper = ({
   const [isZoomed, setIsZoomed] = useState(false);
   const isZoomedValue = useSharedValue(false);
 
+  useEffect(() => {
```

Comment:
can this be handle in the outer component? e.g., 
`onZoomIn={() => StatusBar.setHidden(true)}`

I believe this is breaking SRP and OCP.

Reply:
I'm not too sure if this is the best approach since we also use `hideStatusBar` in other places in this component to alter the height and y translation, which may cause inconsistent UI if the consumer forgets to hide the status bar in the zoom callbacks. What is SRP and OCP, and why would they be breaking?

Reply:
single responsibility principle - the class is responsible for more than one thing
open-closed principle - we need to modify the class to obtain an intended behavior, not just extend this (in this case, via props).

(S and O in SOLID rules) 

But ok, this is only one thing, maybe let it be in the current state.